Automattic/wp-calypso
# Editor: Use Redux state for Post Formats

This PR is a continuation of #8402, which migrated the legacy `lib/post-formats` Flux store to Redux. This PR:

* Modifies the post formats to use the new Redux state
* ES6ifies the editor post formats components
* Fixes several ESLint warnings.
* Removes the unused legacy `PostFormatsStore` Flux store.
* Removes the unused legacy `PostFormatsData` data component.

To test:

* Checkout this branch
* Go to `http://calypso.localhost:3000/post/$site/`, where `$site` is a site supports one or more additional post formats.
* Create a new post with a different post format and save it.
* Open the edit page of the newly created post.
* In all of the above cases, play around with the post formats and verify they work as expected (change, save and retrieve accordingly).
